src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -458,15 +458,47 @@ module Base = struct
          in
          Boolean.Assert.is_true correct_coinbase_stack)
     in
-    let%bind root =
+    let account_creation_amount_var =
+      Amount.Checked.of_fee
+        Fee.(var_of_t Coda_compile_config.account_creation_fee)
+    in
+    let%bind receiver =
+      (* A stake delegation only uses the sender *)
+      Public_key.Compressed.Checked.if_ is_stake_delegation
+        ~then_:sender_compressed ~else_:payload.body.public_key
+    in
+    (* we explicitly set the public_key because it could be zero if the account is new *)
+    let%bind root_after_receiver_update =
+      (* This update should be a no-op in the stake delegation case *)
+      Frozen_ledger_hash.modify_account_recv root receiver
+        ~f:(fun ~is_empty_and_writeable account ->
+          let%map balance =
+            (* receiver_increase will be zero in the stake delegation case *)
+            let%bind receiver_amount =
+              let%bind amount_for_new_account =
+                Amount.Checked.sub receiver_increase
```

Comment:
This also asserts that `receiver_increase` is greater than `amount_creation_amount`. We probably do not want to do that. Instead we can use `sub_flagged` and assert that the subtraction did not underflow in the case that we are creating a new amount.